open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/inmemory-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/inmemory/repositoryconnector/InMemoryEntityNeighbourhood.java

```diff
@@ -0,0 +1,349 @@
+    private Map<String,Set<String>> relationshipToEntities= new HashMap<String,Set<String>>();
+
+    /**
+     * Contructor
+     * @param repositoryValidator repository validator
+     * @param entityStore          entity store
+     * @param relationshipStore relationship store
+     * @param rootEntityGUID the starting point of the query.
+     * @param entityTypeGUIDs list of entity types to include in the query results.  Null means include
+     *                          all entities found, irrespective of their type.
+     * @param relationshipTypeGUIDs list of relationship types to include in the query results.  Null means include
+     *                          all entities found, irrespective of their type.
+     * @param limitResultsByStatus By default, relationships in all statuses are returned.  However, it is possible
+     *                             to specify a list of statuses (eg ACTIVE) to restrict the results to.  Null means all
+     *                             status values.
+     * @param limitResultsByClassification List of classifications that must be present on all returned entities.
+     * @param level the number of the relationships out from the starting entity that the query will traverse to
+     *              gather results.
+     * @throws RepositoryErrorException there is a problem communicating with the metadata repository where
+     *                                  the metadata collection is stored.
+     * @throws InvalidParameterException one of the parameters is invalid or null.
+     */
+    public InMemoryEntityNeighbourhood(
+                                       OMRSRepositoryValidator repositoryValidator,
+                                       Map<String, EntityDetail>   entityStore,
+                                       Map<String, Relationship>   relationshipStore,
+                                       String               rootEntityGUID,
+                                       List<String>         entityTypeGUIDs,
+                                       List<String>         relationshipTypeGUIDs,
+                                       List<InstanceStatus> limitResultsByStatus,
+                                       List<String>         limitResultsByClassification,
+                                       int                  level) throws RepositoryErrorException, InvalidParameterException
+    {
+        this.repositoryValidator =repositoryValidator;
+        this.entityStore = entityStore;
+        this.relationshipStore = relationshipStore;
+        this.rootEntityGUID = rootEntityGUID;
+        this.entityTypeGUIDs =entityTypeGUIDs;
+        this.relationshipTypeGUIDs = relationshipTypeGUIDs;
+        this.limitResultsByStatus =limitResultsByStatus;
+        this.limitResultsByClassification=limitResultsByClassification;
+        // limit the level
+        if (level <1 || level >10 ) {
```

Comment:
Why 10?

Reply:
May be 10 is too low - I was thinking of a throttle point in case it went on for ever - I will change to 100.

Reply:
amended